tests/versioned/aws-sdk.tap.js

```
@@ -9,71 +9,89 @@ const tap = require('tap')
-    helper.registerInstrumentation({
-      moduleName: 'aws-sdk',
-      type: 'conglomerate',
-      onRequire: require('../../lib/instrumentation')
+    server = createEmptyResponseServer()
+    server.listen(0, () => {
+      helper = utils.TestAgent.makeInstrumented()
+      helper.registerInstrumentation({
+        moduleName: 'aws-sdk',
+        type: 'conglomerate',
+        onRequire: require('../../lib/instrumentation')
+      })
+      AWS = require('aws-sdk')
+      AWS.config.update({region: 'us-east-1'})
+
+      endpoint = `http://localhost:${server.address().port}`
+      done()
     })
-    AWS = require('aws-sdk')
-    done()
   })
 
   t.afterEach((done) => {
+    server.close()
+    server = null
+
     helper && helper.unload()
     AWS = null
     done()
   })
 
-  t.test('should mark requests to be dt-disabled', (t) => {
-    const https = require('https')
-    sinon.spy(https, 'request')
+  t.test('should mark requests to be dt-disabled', {skip: true}, (t) => {
+    // http because we've changed endpoint to be http
+    const http = require('http')
+    sinon.spy(http, 'request')
     t.tearDown(() => {
       // `afterEach` runs before `tearDown`, so the sinon spy may have already
       // been removed.
-      if (https.request.restore) {
-        https.request.restore()
+      if (http.request.restore) {
+        http.request.restore()
       }
     })
 
-    AWS.config.update({
-      region: 'region',
-      credentials: new AWS.CognitoIdentityCredentials({
-        IdentityPoolId: 'foobar'
-      })
-    })
-
     const s3 = new AWS.S3({
       apiVersion: '2006-03-01',
+      credentials: FAKE_CREDENTIALS,
+      endpoint: endpoint,
+      // allows using generic endpoint, instead of needing a
+      // bucket.endpoint server setup.
+      s3ForcePathStyle: true,
       params: {Bucket: 'bucket'}
     })
-    s3.listObjects({Delimiter: '/'}, () => {})
+    s3.listObjects({Delimiter: '/'}, (err) => {
```

Was previously setup to error out every time with a full round-trip and ignore the error. 

Now returns successfully from the stub, so we validate if there's an unexpected error generated by the sdk.